contracts/masset/Masset.sol

```diff
@@ -656,10 +832,48 @@ contract Masset is
         outputMinusFee = _bAssetQuantity.sub(feeAmount);
     }
 
+    /**
+     * vaultBalanceSum = totalSupply + 'surplus'
+     * maxCache = vaultBalanceSum * (cacheSize / 1e18)
+     * surplus is simply surplus, to reduce SLOADs
+     */
+    struct Cache {
+        uint256 vaultBalanceSum;
+        uint256 maxCache;
+        uint256 surplus;
+    }
+
+    /**
+     * @dev Gets the supply and cache details for the mAsset, taking into account the surplus
+     * @return Cache containing (tracked) sum of vault balances, ideal cache size and surplus
+     */
+    function _getCacheDetails() internal view returns (Cache memory) {
+        uint256 _surplus = surplus;
+        uint256 sum = totalSupply().add(_surplus);
+        return Cache(sum, sum.mulTruncate(cacheSize), _surplus);
+    }
+
     /***************************************
                     STATE
     ****************************************/
 
+    /**
+      * @dev Sets the MAX cache size for each bAsset. The cache will actually revolve around
+      *      _cacheSize * totalSupply / 2 under normal circumstances.
+      * @param _cacheSize Maximum percent of total mAsset supply to hold for each bAsset
+      */
+    function setCacheSize(uint256 _cacheSize)
+        external
+        onlyGovernance
+    {
+        require(_cacheSize <= 2e17, "Must be <= 20%");
+
+        cacheSize = _cacheSize;
+
```

Comment:
A reduction in cache size might temporily violate the invariant that the cached value is <= cacheSize * totalSupply. Is this intended? Or should it trigger a deposit into the lending pool?

Reply:
The next time an action (mint,swap,redeem) is called, it should even itself out based on the new cache size.

Reply:
Yeah, I agree. This observation is mostly to make sure you're aware that there may be short periods of time where the invariant does not hold. This is mostly important if you're assuming the invariant in other places of the code.